### PR TITLE
Develop dashboard

### PR DESCRIPTION
**Pull request by Codefeak**
Dashboard combined with Leftpannel and other containers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,14 +27,14 @@
     "eject": "react-scripts eject"
   },
   "devDependencies": {
+    "csslint": "1.0.5",
```
Comment (trungh13) [inline package.json]: change to right order so no changes these lines

File: src/containers/Dashboard/components/EventPanel/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import * as actions from '../../../../store/actions/index';
+import { getEvent } from '../../../../store/reducers/index';
+
+class EventPanel extends Component {
+  componentWillMount() {
```
Comment (trungh13) [inline src/containers/Dashboard/components/EventPanel/index.js]: componentDidMount better right ? @Nizar-Rahme

File: src/index.js [hunk 3]
```diff
@@ -5,7 +5,6 @@ import { createStore, applyMiddleware, compose } from 'redux';
 import { Provider } from 'react-redux';
 import thunk from 'redux-thunk';
 import { BrowserRouter as Router } from 'react-router-dom';
-
 import reducers from './store/reducers';
```
Comment (trungh13) [inline src/index.js]: still need space

File: src/store/actions/index.js [hunk 4]
```diff
@@ -10,6 +17,39 @@ export const fetchUser = () => dispatch => axios.get('https://integrify.network/
...
+  axios.get('https://integrify.network/api/dashboard/assignments').then((res) => {
+    dispatch({
+      type: FETCH_ASSIGNMENT,
+      payload: res.data,
+    });
+  });
+};
+
+// wait data to load
+export const setEventsLoading = () => ({
```
Comment (trungh13) [inline src/store/actions/index.js]: need a component and action for general loading to reuse then the separately

File: src/containers/Dashboard/components/EventPanel/index.js (hunk 2 above)
Comment (trungh13) [reply src/containers/Dashboard/components/EventPanel/index.js]: https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/
take a look

Comment (Codefeak) [reply src/containers/Dashboard/components/EventPanel/index.js]: changed it to componentDidMount

File: src/store/actions/index.js (hunk 4 above)
Comment (Codefeak) [reply src/store/actions/index.js]: removed Events Loading for now

File: package.json (hunk 1 above)
Comment (Codefeak) [reply package.json]: Should I change something or not??

File: src/index.js (hunk 3 above)
Comment (Codefeak) [reply src/index.js]: not clear what ? should i keep that space or delete it or add one more??

File: src/store/reducers/eventsReducer.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    case GET_EVENT_LIST:
+      return {
+        ...state,
+        events: action.payload,
+        loading: false,
+      };
+    case EVENTS_LOADING:
+      return {
+        ...state,
+        loading: true,
```
Comment (Nizar-Rahme) [inline src/store/reducers/eventsReducer.js]: we should handle `loading` separately for all api requests

**Pull request merged**